sympy/physics/mechanics/kane.py

```diff
@@ -15,48 +15,87 @@
     q, coordinates : Matrix, shape(n, 1)
-        Matrices of the generalized coordinates.
+        The generalized coordinates. These are ordered as [q_s, q_r]^T.
+    o, num_dependent_coordinates : integer
+        The number of dependent coordinates.
+    q_s, independent_coodrinates : Matrix, shape(n-o, 1)
+        The independent generalized coordinates.
+    q_r, dependent_coodrinates : Matrix, shape(o, 1)
+        The dependent generalized coordinates.
     u, speeds : Matrix, shape(n, 1)
-        Matrices of the generalized speeds. For a non-holonomic system these
-        are ordered as [u_s, u_r].
-    bodies, bodylist : iterable
-        Iterable of Particle and RigidBody objects in the system.
-    loads, forcelist : iterable
-        Iterable of (Point, vector) or (ReferenceFrame, vector) tuples
-        describing the forces and torques on the system.
+        The generalized speeds. For a nonholomic system these are ordered as
+        [u_s, u_r]^T.
+    p, num_indepenent_speeds : integer
+        The number of independen speeds.
+    u_s, independent_speeds : Matrix, shape(p, 1)
+        The independent generalized speeds.
+    m, num_dependent_speeds : integer
+        The number of dependent speeds.
+    u_r, dependent_speeds : Matrix, shape(m, 1)
+        The dependent generalized speeds.
+    holonomic : boolean
+        True if the system is a holonomic system.
+    nonholonomic : boolean
+        True if the system is a nonholonomic system.
+    bodies, bodylist : list
+        The Particle and RigidBody objects in the system.
+    loads, forcelist : list
+        The (Point, Vector) or (ReferenceFrame, Vector) tuples describing the
+        forces and torques on the system.
     auxiliary : Matrix
         If applicable, the set of auxiliary Kane's equations used to solve for
         non-contributing forces.
+    fr, generalized_active_forces : Matrix, shape(n, 1)
+        The generalized active forces acting on the system.
+    frstar, generalized_inertia_forces : Matrix, shape(n, 1)
+        The genealized inertia forcies acting on the system.
     mass_matrix : Matrix, shape(n, n)
-        The "mass matrix" of the dynamic differential equations.
+        The "mass matrix" of the implicit dynamic differential equations.
     forcing : Matrix, shape(n, 1)
         The right hand side or "forcing vector" of the implicit dynamic
         differential equations.
-    mass_matrix_full : Matrix
-        The "mass matrix" for the u's and q's
-    forcing_full : Matrix
-        The "forcing vector" for the u's and q's
+    mass_matrix_full : Matrix, shape(2n, 2n)
+        The "mass matrix" of the full first order form of the implicit dynamic
+        and kinematic differential equations.
+    forcing_full : Matrix, shape(2n, 1)
+        The right hand side or "forcing vector" of the full first order form of
+        the implicit dynamic and kinematic differential equations.
```

Comment:
I think `self.rhs()` should be mentioned somewhere in the docstring as the method of obtaining the equations of motion in explicit form. I don't know exactly where it would go though because it is not an attribute like the other but rather a method so that it's potentially long calculations don't bog down code unless it is required.